phase/extender.go

```diff
@@ -387,16 +387,35 @@ func (e *Extender) dockerfileFor(kind, extID string) (*extend.Dockerfile, error)
 		args = config.Run.Args
 	}
 
+	contextDir, err := e.contextDirFor(kind, extID)
+	if err != nil {
+		return nil, err
+	}
+
 	return &extend.Dockerfile{
 		ExtensionID: extID,
 		Path:        dockerfilePath,
 		Args:        args,
+		ContextDir:  contextDir,
 	}, nil
 }
 
-func (e *Extender) extendOptions() extend.Options {
+func (e *Extender) contextDirFor(kind, extID string) (string, error) {
+	sharedContextDir := filepath.Join(e.GeneratedDir, launch.EscapeID(extID), buildpack.SharedContextDir)
+	kindContextDir := filepath.Join(e.GeneratedDir, launch.EscapeID(extID), fmt.Sprintf("%s.%s", buildpack.SharedContextDir, kind))
+
+	for _, dir := range []string{kindContextDir, sharedContextDir} {
+		if s, err := os.Stat(dir); err == nil && s.IsDir() {
+			return dir, nil
+		}
+	}
+
+	return e.AppDir, nil
+}
+
+func (e *Extender) extendOptions(dockerfile extend.Dockerfile) extend.Options {
 	return extend.Options{
-		BuildContext: e.AppDir,
+		BuildContext: dockerfile.ContextDir,
```

Comment:
We should default to the app dir if no context dir is provided

Reply:
Defaulted in https://github.com/buildpacks/lifecycle/blob/a3fc1ef8508c88c1cdc37890b9224096241ebe66/phase/extender.go#L376.